File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/workspace_object_permission_summary.py
# coding: utf-8
# Copyright (c) 2016, 2026, Oracle and/or its affiliates.  All rights reserved.
# This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.


from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class WorkspaceObjectPermissionSummary(object):
    """
    Information about a workspace object permission.
    """

    #: A constant which can be used with the grantee_type property of a WorkspaceObjectPermissionSummary.
    #: This constant has a value of "USER"
    GRANTEE_TYPE_USER = "USER"

    #: A constant which can be used with the grantee_type property of a WorkspaceObjectPermissionSummary.
    #: This constant has a value of "ROLE"
    GRANTEE_TYPE_ROLE = "ROLE"

    #: A constant which can be used with the grantee_type property of a WorkspaceObjectPermissionSummary.
    #: This constant has a value of "GROUP"
    GRANTEE_TYPE_GROUP = "GROUP"

    #: A constant which can be used with the grantee_permissions property of a WorkspaceObjectPermissionSummary.
    #: This constant has a value of "READ"
    GRANTEE_PERMISSIONS_READ = "READ"

    #: A constant which can be used with the grantee_permissions property of a WorkspaceObjectPermissionSummary.
    #: This constant has a value of "USE"
    GRANTEE_PERMISSIONS_USE = "USE"

    #: A constant which can be used with the grantee_permissions property of a WorkspaceObjectPermissionSummary.
    #: This constant has a value of "MANAGE"
    GRANTEE_PERMISSIONS_MANAGE = "MANAGE"

    #: A constant which can be used with the grantee_permissions property of a WorkspaceObjectPermissionSummary.
    #: This constant has a value of "ADMIN"
    GRANTEE_PERMISSIONS_ADMIN = "ADMIN"

    def __init__(self, **kwargs):
        """
        Initializes a new WorkspaceObjectPermissionSummary object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param grantee:
            The value to assign to the grantee property of this WorkspaceObjectPermissionSummary.
        :type grantee: str

        :param grantee_name:
            The value to assign to the grantee_name property of this WorkspaceObjectPermissionSummary.
        :type grantee_name: str

        :param grantee_type:
            The value to assign to the grantee_type property of this WorkspaceObjectPermissionSummary.
            Allowed values for this property are: "USER", "ROLE", "GROUP", 'UNKNOWN_ENUM_VALUE'.
            Any unrecognized values returned by a service will be mapped to 'UNKNOWN_ENUM_VALUE'.
        :type grantee_type: str

        :param grantee_permissions:
            The value to assign to the grantee_permissions property of this WorkspaceObjectPermissionSummary.
            Allowed values for items in this list are: "READ", "USE", "MANAGE", "ADMIN", 'UNKNOWN_ENUM_VALUE'.
            Any unrecognized values returned by a service will be mapped to 'UNKNOWN_ENUM_VALUE'.
        :type grantee_permissions: list[str]

        :param is_permissions_inheritable:
            The value to assign to the is_permissions_inheritable property of this WorkspaceObjectPermissionSummary.
        :type is_permissions_inheritable: bool

        """
        self.swagger_types = {
            'grantee': 'str',
            'grantee_name': 'str',
            'grantee_type': 'str',
            'grantee_permissions': 'list[str]',
            'is_permissions_inheritable': 'bool'
        }

        self.attribute_map = {
            'grantee': 'grantee',
            'grantee_name': 'granteeName',
            'grantee_type': 'granteeType',
            'grantee_permissions': 'granteePermissions',
            'is_permissions_inheritable': 'isPermissionsInheritable'
        }

        self._grantee = None
        self._grantee_name = None
        self._grantee_type = None
        self._grantee_permissions = None
        self._is_permissions_inheritable = None

    @property
    def grantee(self):
        """
        **[Required]** Gets the grantee of this WorkspaceObjectPermissionSummary.
        The OCID of user/group and name in case of role.


        :return: The grantee of this WorkspaceObjectPermissionSummary.
        :rtype: str
        """
        return self._grantee

    @grantee.setter
    def grantee(self, grantee):
        """
        Sets the grantee of this WorkspaceObjectPermissionSummary.
        The OCID of user/group and name in case of role.


        :param grantee: The grantee of this WorkspaceObjectPermissionSummary.
        :type: str
        """
        self._grantee = grantee

    @property
    def grantee_name(self):
        """
        Gets the grantee_name of this WorkspaceObjectPermissionSummary.
        The simplified name of the grantee.


        :return: The grantee_name of this WorkspaceObjectPermissionSummary.
        :rtype: str
        """
        return self._grantee_name

    @grantee_name.setter
    def grantee_name(self, grantee_name):
        """
        Sets the grantee_name of this WorkspaceObjectPermissionSummary.
        The simplified name of the grantee.


        :param grantee_name: The grantee_name of this WorkspaceObjectPermissionSummary.
        :type: str
        """
        self._grantee_name = grantee_name

    @property
    def grantee_type(self):
        """
        **[Required]** Gets the grantee_type of this WorkspaceObjectPermissionSummary.
        The type of grantee.

        Allowed values for this property are: "USER", "ROLE", "GROUP", 'UNKNOWN_ENUM_VALUE'.
        Any unrecognized values returned by a service will be mapped to 'UNKNOWN_ENUM_VALUE'.


        :return: The grantee_type of this WorkspaceObjectPermissionSummary.
        :rtype: str
        """
        return self._grantee_type

    @grantee_type.setter
    def grantee_type(self, grantee_type):
        """
        Sets the grantee_type of this WorkspaceObjectPermissionSummary.
        The type of grantee.


        :param grantee_type: The grantee_type of this WorkspaceObjectPermissionSummary.
        :type: str
        """
        allowed_values = ["USER", "ROLE", "GROUP"]
        if not value_allowed_none_or_none_sentinel(grantee_type, allowed_values):
            grantee_type = 'UNKNOWN_ENUM_VALUE'
        self._grantee_type = grantee_type

    @property
    def grantee_permissions(self):
        """
        **[Required]** Gets the grantee_permissions of this WorkspaceObjectPermissionSummary.
        The selected permissions for a workspace object.

        Allowed values for items in this list are: "READ", "USE", "MANAGE", "ADMIN", 'UNKNOWN_ENUM_VALUE'.
        Any unrecognized values returned by a service will be mapped to 'UNKNOWN_ENUM_VALUE'.


        :return: The grantee_permissions of this WorkspaceObjectPermissionSummary.
        :rtype: list[str]
        """
        return self._grantee_permissions

    @grantee_permissions.setter
    def grantee_permissions(self, grantee_permissions):
        """
        Sets the grantee_permissions of this WorkspaceObjectPermissionSummary.
        The selected permissions for a workspace object.


        :param grantee_permissions: The grantee_permissions of this WorkspaceObjectPermissionSummary.
        :type: list[str]
        """
        allowed_values = ["READ", "USE", "MANAGE", "ADMIN"]
        if grantee_permissions:
            grantee_permissions[:] = ['UNKNOWN_ENUM_VALUE' if not value_allowed_none_or_none_sentinel(x, allowed_values) else x for x in grantee_permissions]
        self._grantee_permissions = grantee_permissions

    @property
    def is_permissions_inheritable(self):
        """
        Gets the is_permissions_inheritable of this WorkspaceObjectPermissionSummary.
        Property to determine if permission is inheritable or not. This is applicable only on workspace folders not on files.


        :return: The is_permissions_inheritable of this WorkspaceObjectPermissionSummary.
        :rtype: bool
        """
        return self._is_permissions_inheritable

    @is_permissions_inheritable.setter
    def is_permissions_inheritable(self, is_permissions_inheritable):
        """
        Sets the is_permissions_inheritable of this WorkspaceObjectPermissionSummary.
        Property to determine if permission is inheritable or not. This is applicable only on workspace folders not on files.


        :param is_permissions_inheritable: The is_permissions_inheritable of this WorkspaceObjectPermissionSummary.
        :type: bool
        """
        self._is_permissions_inheritable = is_permissions_inheritable

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other

File: aidp-java-client/src/main/java/com/oracle/aidataplatform/dp/model/WorkspaceObjectPermissionSummary.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;



/**
 * Information about a workspace object permission.
**/
@jakarta.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=WorkspaceObjectPermissionSummary.Builder.class)

public final class WorkspaceObjectPermissionSummary  {
    @Deprecated
    @java.beans.ConstructorProperties({"grantee", "granteeName", "granteeType", "granteePermissions", "isPermissionsInheritable"})
    public WorkspaceObjectPermissionSummary(String grantee, String granteeName, GranteeType granteeType, java.util.List<WorkspaceObjectPrivilege> granteePermissions, Boolean isPermissionsInheritable) {
        super();
        this.grantee = grantee;
        this.granteeName = granteeName;
        this.granteeType = granteeType;
        this.granteePermissions = granteePermissions;
        this.isPermissionsInheritable = isPermissionsInheritable;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
                /**
     * The OCID of user/group and name in case of role.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("grantee")
private String grantee;

        /**
         * The OCID of user/group and name in case of role.
         * @param grantee the value to set
         * @return this builder
         **/
        

public Builder grantee(String grantee) {
    this.grantee = grantee;
    return this;
}
            /**
     * The simplified name of the grantee.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("granteeName")
private String granteeName;

        /**
         * The simplified name of the grantee.
         * @param granteeName the value to set
         * @return this builder
         **/
        

public Builder granteeName(String granteeName) {
    this.granteeName = granteeName;
    return this;
}
            /**
     * The type of grantee.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("granteeType")
private GranteeType granteeType;

        /**
         * The type of grantee.
         * @param granteeType the value to set
         * @return this builder
         **/
        

public Builder granteeType(GranteeType granteeType) {
    this.granteeType = granteeType;
    return this;
}
            /**
     * The selected permissions for a workspace object.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("granteePermissions")
private java.util.List<WorkspaceObjectPrivilege> granteePermissions;

        /**
         * The selected permissions for a workspace object.
         * @param granteePermissions the value to set
         * @return this builder
         **/
        

public Builder granteePermissions(java.util.List<WorkspaceObjectPrivilege> granteePermissions) {
    this.granteePermissions = granteePermissions;
    return this;
}
            /**
     * Property to determine if permission is inheritable or not. This is applicable only on workspace folders not on files.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("isPermissionsInheritable")
private Boolean isPermissionsInheritable;

        /**
         * Property to determine if permission is inheritable or not. This is applicable only on workspace folders not on files.
         * @param isPermissionsInheritable the value to set
         * @return this builder
         **/
        

public Builder isPermissionsInheritable(Boolean isPermissionsInheritable) {
    this.isPermissionsInheritable = isPermissionsInheritable;
    return this;
}


        public WorkspaceObjectPermissionSummary build() {
            WorkspaceObjectPermissionSummary model = new WorkspaceObjectPermissionSummary(this.grantee
                , this.granteeName
                , this.granteeType
                , this.granteePermissions
                , this.isPermissionsInheritable);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(WorkspaceObjectPermissionSummary model) {
                this.grantee(model.getGrantee());
    this.granteeName(model.getGranteeName());
    this.granteeType(model.getGranteeType());
    this.granteePermissions(model.getGranteePermissions());
    this.isPermissionsInheritable(model.getIsPermissionsInheritable());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


        /**
     * The OCID of user/group and name in case of role.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("grantee")
    private final String grantee;

        /**
     * The OCID of user/group and name in case of role.
     * @return the value
     **/
    
    public String getGrantee() {
        return grantee;
    }


        /**
     * The simplified name of the grantee.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("granteeName")
    private final String granteeName;

        /**
     * The simplified name of the grantee.
     * @return the value
     **/
    
    public String getGranteeName() {
        return granteeName;
    }

    
        /**
     * The type of grantee.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("granteeType")
    private final GranteeType granteeType;

        /**
     * The type of grantee.
     * @return the value
     **/
    
    public GranteeType getGranteeType() {
        return granteeType;
    }

    
        /**
     * The selected permissions for a workspace object.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("granteePermissions")
    private final java.util.List<WorkspaceObjectPrivilege> granteePermissions;

        /**
     * The selected permissions for a workspace object.
     * @return the value
     **/
    
    public java.util.List<WorkspaceObjectPrivilege> getGranteePermissions() {
        return granteePermissions;
    }


        /**
     * Property to determine if permission is inheritable or not. This is applicable only on workspace folders not on files.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("isPermissionsInheritable")
    private final Boolean isPermissionsInheritable;

        /**
     * Property to determine if permission is inheritable or not. This is applicable only on workspace folders not on files.
     * @return the value
     **/
    
    public Boolean getIsPermissionsInheritable() {
        return isPermissionsInheritable;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("WorkspaceObjectPermissionSummary(");
        sb.append("grantee=").append(String.valueOf(this.grantee));
        sb.append(", granteeName=").append(String.valueOf(this.granteeName));
        sb.append(", granteeType=").append(String.valueOf(this.granteeType));
        sb.append(", granteePermissions=").append(String.valueOf(this.granteePermissions));
        sb.append(", isPermissionsInheritable=").append(String.valueOf(this.isPermissionsInheritable));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof WorkspaceObjectPermissionSummary)) {
            return false;
        }

        WorkspaceObjectPermissionSummary other = (WorkspaceObjectPermissionSummary) o;
        return java.util.Objects.equals(this.grantee, other.grantee) &&
            java.util.Objects.equals(this.granteeName, other.granteeName) &&
            java.util.Objects.equals(this.granteeType, other.granteeType) &&
            java.util.Objects.equals(this.granteePermissions, other.granteePermissions) &&
            java.util.Objects.equals(this.isPermissionsInheritable, other.isPermissionsInheritable);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.grantee == null ? 43 : this.grantee.hashCode());
        result = (result * PRIME) + (this.granteeName == null ? 43 : this.granteeName.hashCode());
        result = (result * PRIME) + (this.granteeType == null ? 43 : this.granteeType.hashCode());
        result = (result * PRIME) + (this.granteePermissions == null ? 43 : this.granteePermissions.hashCode());
        result = (result * PRIME) + (this.isPermissionsInheritable == null ? 43 : this.isPermissionsInheritable.hashCode());
        return result;
    }


}
